x-pack/plugins/cases/server/common/types/case.ts

```diff
@@ -48,8 +48,11 @@ export interface CasePersistedAttributes {
   updated_at: string | null;
   updated_by: User | null;
   category?: string | null;
+  customFields?: CasePersistedCustomFields;
```

Comment:
Instead of importing here from the domain types we can do 

```
customFields: Array<{ key: string; type: string; value: unknown }>
```

as we do in `x-pack/plugins/cases/server/common/types/connectors.ts` or `x-pack/plugins/cases/server/common/types/user_actions.ts`. We want to decouple as much as possible the domain types with the types used by the services. Feel free to cast in the transform function (as we do here `x-pack/plugins/cases/server/services/user_actions/transform.ts`) if needed.